test/common/upstream/health_checker_impl_test_utils.cc

```diff
@@ -75,8 +75,8 @@ void HttpHealthCheckerImplTestBase::expectClientCreate(size_t index) {
 // This is needed to put expectations in LIFO order. The unit tests use inSequence, which makes
 // expectations FIFO.
 void TcpHealthCheckerImplTestBase::expectSessionCreate() {
-  timeout_timer_ = new Event::MockTimer(&dispatcher_);
-  interval_timer_ = new Event::MockTimer(&dispatcher_);
+  timeout_timer_ = new NiceMock<Event::MockTimer>(&dispatcher_);
```

Comment:
thanks! assuming in a later TODO there will be more TCP fixups?

Reply:
Yup! One I know I'm going to do for maintainability is get rid of these base classes if so permits so not tightly coupled with unit tests.